src/utils/token-standard/v2/settlement-factory.ts

```
@@ -0,0 +1,517 @@
+}
+
+function requireText(value: unknown, field: string): string {
+  if (typeof value !== 'string') {
+    inputInvalid(`${field} must be a string.`, { field, value });
+  }
+  return value;
+}
+
+function parseDecimalText(value: unknown, field: string): { readonly text: string; readonly sign: -1 | 0 | 1 } {
+  const text = requireText(value, field);
+  const match = /^(-?)(\d{1,28})(?:\.(\d{1,10}))?$/.exec(text);
```

ESLint

[error] Move this regular expression to module scope to avoid re-compilation on every call.

(e18e/prefer-static-regex)